expression/schema.go

```diff
@@ -107,6 +107,9 @@ func (s *Schema) FindColumnAndIndex(astCol *ast.ColumnName) (*Column, int, error
 			if idx == -1 {
 				idx = i
 			} else {
+				if col.IsInsertSelect && s.Columns[idx].IsInsertSelect {
```

Comment:
This looks not right, check out this case:
In MySQL:
```
mysql> create table t2(f1 int primary key, f2 int);
Query OK, 0 rows affected (0.00 sec)

mysql> insert into t2 values(1,2),(2,1);
Query OK, 2 rows affected (0.00 sec)
Records: 2  Duplicates: 0  Warnings: 0

mysql> INSERT INTO t2 (f1, f2) SELECT f2 f1, f1 FROM t2 src ON DUPLICATE KEY UPDATE f1 = 10 + src.f1;
Query OK, 4 rows affected (0.00 sec)
Records: 2  Duplicates: 2  Warnings: 0

mysql> select * from t2;
+----+------+
| f1 | f2   |
+----+------+
| 11 |    1 |
| 12 |    2 |
+----+------+
```
while in TiDB after applying your patch:
```
mysql> create table t2(f1 int primary key, f2 int);
Query OK, 0 rows affected (0.00 sec)

mysql> insert into t2 values(1,2),(2,1);
Query OK, 2 rows affected (0.00 sec)
Records: 2  Duplicates: 0  Warnings: 0

mysql> INSERT INTO t2 (f1, f2) SELECT f2 f1, f1 FROM t2 src ON DUPLICATE KEY UPDATE f1 = 10 + src.f1;
Query OK, 4 rows affected (0.00 sec)
Records: 2  Duplicates: 2  Warnings: 0

mysql> select * from t2;
+----+------+
| f1 | f2   |
+----+------+
| 11 |    2 |
| 12 |    1 |
+----+------+
```

Seems like MySQL would resolve the `src.f1` to the column which has original name `f1` in this query, not the column which has alias name `f1` while its original column name is `f2`.

Reply:
this seems anther bug.
```sql
 drop table if exists t2;
         create table t2(f1 int primary key, f2 int);
         insert into t2 values(1,2),(2,1);
INSERT INTO t2 (f1, f2) SELECT f2 f1, f1 f2 FROM t2 src ON DUPLICATE KEY UPDATE f1 = 10 + src.f1;
         select * from t2;
```
Resutlt in mysql is `11	2 ;12	1`
While in tidb-master is `11	1; 12	2`

Reply:
Could you please help fix this as well?

Reply:
```sql
INSERT INTO t2 (f1, f2)
   SELECT src.f1 f2, src.f2 f1 FROM t2 src where src.f2=1
  ON DUPLICATE KEY UPDATE f1 = 100 + src.f2
```
This resolver is so strange.
There are three `src.f2`, first and second are same.But the second is not.
I will try to fix it.